geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnection.java

```diff
@@ -158,10 +158,10 @@ public static void emptyCommBufferPool() {
 
   // IMPORTANT: if new messages are added change setHandshake to initialize them
   // to the correct Version for serializing to the client
-  private Message requestMsg = new Message(2, Version.CURRENT);
-  private Message replyMsg = new Message(1, Version.CURRENT);
-  private Message responseMsg = new Message(1, Version.CURRENT);
-  private Message errorMsg = new Message(1, Version.CURRENT);
+  private Message requestMsg;
```

Comment:
These could be changed to MessageFromClient & MessageFromServer.  I think that would improve the clarity of these declarations.  Then you could remove the casts to these classes that you've added in other parts of this file.